File: 프로그래머스/0/180000/181934. 조건 문자열/조건 문자열.py
def solution(ineq, eq, n, m):
    # 두 문자열을 조합하여 하나의 연산자로 취급
    equal = ineq + eq
    
    # 조합된 문자열에 따른 조건 비교 수행
    if equal == ">=":
        return 1 if n >= m else 0
    elif equal == "<=":
        return 1 if n <= m else 0
    elif equal == ">!":
        return 1 if n > m else 0
    elif equal == "<!":
        return 1 if n < m else 0
    return 0

File: 프로그래머스/0/180000/181934. 조건 문자열/조건 문자열.java
class Solution {
    public int solution(String ineq, String eq, int n, int m) {
        String equal = ineq + eq;
        
        if (equal.equals(">=")) {
            return n >= m ? 1 : 0;
        } else if (equal.equals("<=")) {
            return n <= m ? 1 : 0;
        } else if (equal.equals(">!")) {
            return n > m ? 1 : 0;
        } else if (equal.equals("<!")) {
            return n < m ? 1 : 0;
        }
        return 0;
    }
}


/**
class Solution {
    public int solution(String ineq, String eq, int n, int m) {
        return (ineq.equals(">") && eq.equals("=") && n >= m) ||
               (ineq.equals("<") && eq.equals("=") && n <= m) ||
               (ineq.equals(">") && eq.equals("!") && n > m) ||
               (ineq.equals("<") && eq.equals("!") && n < m) ? 1 : 0;
    }
}
**/
